cdp-agentkit-core/typescript/src/action_providers/twitter/twitterActionProvider.ts

```diff
@@ -0,0 +1,207 @@
+import { z } from "zod";
+import { ActionProvider } from "../action_provider";
+import { CreateAction } from "../action_decorator";
+import { TwitterApi, TwitterApiTokens } from "twitter-api-v2";
+import { Network } from "../../wallet_providers/wallet_provider";
+import {
+  TwitterAccountDetailsSchema,
+  TwitterAccountMentionsSchema,
+  TwitterPostTweetSchema,
+  TwitterPostTweetReplySchema,
+} from "./schemas";
+
+/**
+ * Configuration options for the TwitterActionProvider.
+ */
+export interface TwitterActionProviderConfig {
+  /**
+   * Twitter API Key
+   */
+  apiKey?: string;
+
+  /**
+   * Twitter API Secret
+   */
+  apiSecret?: string;
+
+  /**
+   * Twitter Access Token
+   */
+  accessToken?: string;
+
+  /**
+   * Twitter Access Token Secret
+   */
+  accessTokenSecret?: string;
+}
+
+/**
+ * TwitterActionProvider is an action provider for Twitter (X) interactions.
+ */
+export class TwitterActionProvider extends ActionProvider {
+  private readonly apiKey: string;
+  private readonly apiSecret: string;
+  private readonly accessToken: string;
+  private readonly accessTokenSecret: string;
+  private readonly client: TwitterApi;
+
+  /**
+   * Constructor for the TwitterActionProvider class.
+   * @param config - The configuration options for the TwitterActionProvider
+   */
+  constructor(config: TwitterActionProviderConfig = {}) {
+    super("twitter", []);
+
+    if (config.apiKey) {
+      this.apiKey = config.apiKey;
+    } else if (process.env.TWITTER_API_KEY) {
+      this.apiKey = process.env.TWITTER_API_KEY;
+    } else {
+      throw new Error("Twitter API Key is not configured.");
+    }
+
+    if (config.apiSecret) {
+      this.apiSecret = config.apiSecret;
+    } else if (process.env.TWITTER_API_SECRET) {
+      this.apiSecret = process.env.TWITTER_API_SECRET;
+    } else {
+      throw new Error("Twitter API Secret is not configured.");
+    }
+
+    if (config.accessToken) {
+      this.accessToken = config.accessToken;
+    } else if (process.env.TWITTER_ACCESS_TOKEN) {
+      this.accessToken = process.env.TWITTER_ACCESS_TOKEN;
+    } else {
+      throw new Error("Twitter Access Token is not configured.");
+    }
+
+    if (config.accessTokenSecret) {
+      this.accessTokenSecret = config.accessTokenSecret;
+    } else if (process.env.TWITTER_ACCESS_TOKEN_SECRET) {
+      this.accessTokenSecret = process.env.TWITTER_ACCESS_TOKEN_SECRET;
+    } else {
+      throw new Error("Twitter Access Token Secret is not configured.");
+    }
+
+    this.client = new TwitterApi({
+      appKey: this.apiKey,
+      appSecret: this.apiSecret,
+      accessToken: this.accessToken,
+      accessSecret: this.accessTokenSecret,
+    } as TwitterApiTokens);
+  }
+
+  /**
+   * Get account details for the currently authenticated Twitter (X) user.
+   */
+  @CreateAction({
+    name: "twitter_account_details",
```

Comment:
```suggestion
    name: "account_details",
```

don't explicitly prefix the action names. I'm going to do some reflection changes that add the action provider name to the action names as a prefix automatically

Reply:
ah roger that!

Reply:
done! rolled back the farcaster prefix too